### ISSUE TITLE
Ignore parts of the URL when checking for uniqueness

### ISSUE THREAD

**Issue by black-snow** [zaproxy/zaproxy#5468]
For the original question see https://groups.google.com/forum/?#!msg/zaproxy-users/6mZzyWvBekg/K3ImfAdiAwAJ

**Is your feature request related to a problem? Please describe.**
We have an application that, sadly, transports the session ID via the URL following the pattern http://example.org/path/action.do;jsessionid=xyz?param=val
What is the same URL to the application (/path/action.do;jsessionid=a and /path/action.do;jsessionid=b) is, of course, not the same URL to ZAP. When I run the spider with our application it never finishes and creates an infinit number of unique nodes which just differ in their session ID.

**Describe the solution you'd like**
If possible, I'd like to tell ZAP that ...;jsessionid=\w+ are the same as the URL without the jsessionid.

**Describe alternatives you've considered**
Excluding all URLs featuring "jsessionid" is not an option. The spider won't detect the bare version in most cases for it carries through one or several IDs.

**Additional context**
-

**Comment by psiinon** [comment]
I dont think the solution is ignoring parts of the URL. I think we need to be able to configure ZAP to handle other characters as URL separators, like ";" in this case.
Note that in this case the URL is using both a semicolon and a question mark, which will be 'challenging'

**Comment by psiinon** [comment]
Actually it looks like we should be able to cope with multiple characters, so having `?;` as 'initial delimiters' and `&?` as 'URL Key Value Pair Separators' will hopefully work.

**Comment by thc202** [comment]
Do we really need to handle this case as a parameter? afaik that's only a hack for the session ID not (additional/other) parameters.

**Comment by psiinon** [comment]
Well, ideally ZAP would understand the URL format correctly, and there are always more apps with weird URL formats.
In what other ways do you think we could handle this?

**Comment by kingthorin** [comment]
@black-snow did you try @psiinon's suggestion?

> Actually it looks like we should be able to cope with multiple characters, so having ?; as 'initial delimiters' and &? as 'URL Key Value Pair Separators' will hopefully work.

**Comment by thc202** [comment]
I don't think that would work the separators apply just to the query component not the whole URL.

> In what other ways do you think we could handle this?

IMO we should allow to ignore that, they are not really parameters it's just used for the session ID.

**Comment by black-snow** [comment]
@kingthorin No, I didn't, but I'm still very interested in this feature - just didn't have the time yet.

**Comment by black-snow** [comment]
> 
> 
> Actually it looks like we should be able to cope with multiple characters, so having `?;` as 'initial delimiters' and `&?` as 'URL Key Value Pair Separators' will hopefully work.

@psiinon Wanted to give this a shot but there ain't no such thing as `initial delimiters`. I also wonder what `Structural Modifiers` are.

Bumping this one because it's getting highly relevant to me again.
